ouroboros-network/src/Ouroboros/Network/Channel.hs

```diff
@@ -43,3 +92,44 @@ withFifosAsChannel fifoPathRead fifoPathWrite action =
     IO.withFile fifoPathWrite IO.WriteMode $ \hndWrite ->
       let channel = handlesAsChannel hndRead hndWrite
        in action channel
+
+
+-- | Make a 'Channel' from a 'Socket'. The socket must be a stream socket
+--- type and status connected.
```

Comment:
Could it be (from `man socket`) as well? mux will send datagrams with of some max length.
> SOCK_SEQPACKET  Provides a sequenced, reliable, two-way connection-based data trans‐
                       mission path for datagrams of fixed maximum length;  a  consumer  is
                       required to read an entire packet with each input system call.

Reply:
I'm not sure what you're getting at exactly. We can't use `SOCK_SEQPACKET` since that doesn't correspond to TCP. We're doing our own framing over TCP. Yes something with `SOCK_SEQPACKET` semantics might work for us, but it doesn't exist on the internet as a standard protocol.